Flaky product test environment setup for multinode-kafka-ssl

https://github.com/trinodb/trino/runs/3872438939

```
2021-10-12T19:42:08.128Z	INFO	environment-listener-1	io.trino.tests.product.launcher.env.EnvironmentListener	Container started: presto-master
2021-10-12T19:42:08.135Z	WARN	main	io.trino.tests.product.launcher.env.Environment	Could not start environment 'multinode-kafka-ssl'
java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at io.trino.tests.product.launcher.env.Environment.tryStart(Environment.java:161)
	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
	at net.jodah.failsafe.RetryPolicyExecutor.lambda$supply$0(RetryPolicyExecutor.java:62)
	at net.jodah.failsafe.Execution.executeSync(Execution.java:129)
	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:67)
	at io.trino.tests.product.launcher.env.Environment.start(Environment.java:114)
	at io.trino.tests.product.launcher.cli.TestRun$Execution.startEnvironment(TestRun.java:240)
	at io.trino.tests.product.launcher.cli.TestRun$Execution.tryExecuteTests(TestRun.java:216)
	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
	at net.jodah.failsafe.TimeoutExecutor.lambda$supply$2(TimeoutExecutor.java:91)
	at net.jodah.failsafe.Execution.executeSync(Execution.java:129)
	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:67)
	at io.trino.tests.product.launcher.cli.TestRun$Execution.call(TestRun.java:199)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.runTest(SuiteRun.java:291)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.executeSuiteTestRun(SuiteRun.java:276)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.lambda$runSuites$2(SuiteRun.java:209)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.lambda$runSuites$3(SuiteRun.java:210)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.runSuites(SuiteRun.java:196)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.call(SuiteRun.java:182)
	at io.trino.tests.product.launcher.cli.SuiteRun$Execution.call(SuiteRun.java:139)
	at io.trino.tests.product.launcher.cli.Commands.runCommand(Commands.java:41)
	at io.trino.tests.product.launcher.cli.SuiteRun.call(SuiteRun.java:104)
	at io.trino.tests.product.launcher.cli.SuiteRun.call(SuiteRun.java:72)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at io.trino.tests.product.launcher.cli.Launcher.run(Launcher.java:75)
	at io.trino.tests.product.launcher.cli.Launcher.main(Launcher.java:65)
Caused by: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at io.trino.tests.product.launcher.env.Environment.tryStart(Environment.java:134)
	... 41 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:315)
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:783)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:65)
	... 13 more


2021-10-12T19:42:08.138Z	INFO	main	io.trino.tests.product.launcher.cli.TestRun	Tests execution completed with code 1
```

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Kafka.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Kafka.java
@@ -21,6 +21,8 @@ import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
 
 import javax.inject.Inject;
 
+import java.time.Duration;
+
 import static io.trino.tests.product.launcher.docker.ContainerUtil.forSelectedPorts;
 import static java.util.Objects.requireNonNull;
 import static org.testcontainers.containers.wait.strategy.Wait.forLogMessage;
@@ -57,7 +59,8 @@ public class Kafka
                 .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
                 .withEnv("ZOOKEEPER_TICK_TIME", "2000")
                 .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
-                .waitingFor(forSelectedPorts(2181));
+                .waitingFor(forSelectedPorts(2181))
+                .withStartupTimeout(Duration.ofMinutes(5));
 
         portBinder.exposePort(container, 2181);
 
@@ -74,7 +77,8 @@ public class Kafka
                 .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1")
                 .withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0")
                 .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
-                .waitingForAll(forSelectedPorts(9092), forLogMessage(".*started \\(kafka.server.KafkaServer\\).*", 1));
+                .waitingForAll(forSelectedPorts(9092), forLogMessage(".*started \\(kafka.server.KafkaServer\\).*", 1))
+                .withStartupTimeout(Duration.ofMinutes(5));
 
         portBinder.exposePort(container, 9092);
 
@@ -89,7 +93,8 @@ public class Kafka
                 .withEnv("SCHEMA_REGISTRY_HOST_NAME", "0.0.0.0")
                 .withEnv("SCHEMA_REGISTRY_LISTENERS", "http://0.0.0.0:" + SCHEMA_REGISTRY_PORT)
                 .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
-                .waitingFor(forSelectedPorts(SCHEMA_REGISTRY_PORT));
+                .waitingFor(forSelectedPorts(SCHEMA_REGISTRY_PORT))
+                .withStartupTimeout(Duration.ofMinutes(5));
 
         portBinder.exposePort(container, SCHEMA_REGISTRY_PORT);
 

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/KafkaSsl.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/KafkaSsl.java
@@ -20,6 +20,7 @@ import org.testcontainers.containers.BindMode;
 
 import javax.inject.Inject;
 
+import java.time.Duration;
 import java.util.List;
 
 import static java.util.Objects.requireNonNull;
@@ -38,7 +39,10 @@ public class KafkaSsl
     @Override
     public void extendEnvironment(Environment.Builder builder)
     {
-        builder.configureContainer(Kafka.KAFKA, container -> container.withEnv("KAFKA_ADVERTISED_LISTENERS", "SSL://kafka:9092")
+        builder.configureContainer(Kafka.KAFKA, container -> container
+                .withStartupAttempts(3)
+                .withStartupTimeout(Duration.ofMinutes(5))
+                .withEnv("KAFKA_ADVERTISED_LISTENERS", "SSL://kafka:9092")
                 .withEnv("KAFKA_SSL_KEYSTORE_FILENAME", "kafka.broker1.keystore")
                 .withEnv("KAFKA_SSL_KEYSTORE_CREDENTIALS", "broker1_keystore_creds")
                 .withEnv("KAFKA_SSL_KEYSTORE_TYPE", "JKS")
@@ -50,7 +54,10 @@ public class KafkaSsl
                 .withEnv("KAFKA_SSL_CLIENT_AUTH", "required")
                 .withEnv("KAFKA_SECURITY_INTER_BROKER_PROTOCOL", "SSL")
                 .withClasspathResourceMapping("docker/presto-product-tests/conf/environment/multinode-kafka-ssl/secrets", "/etc/kafka/secrets", BindMode.READ_ONLY));
-        builder.configureContainer(Kafka.SCHEMA_REGISTRY, container -> container.withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", "SSL://kafka:9092")
+        builder.configureContainer(Kafka.SCHEMA_REGISTRY, container -> container
+                .withStartupAttempts(3)
+                .withStartupTimeout(Duration.ofMinutes(5))
+                .withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", "SSL://kafka:9092")
                 .withEnv("SCHEMA_REGISTRY_KAFKASTORE_SECURITY_PROTOCOL", "SSL")
                 .withEnv("SCHEMA_REGISTRY_KAFKASTORE_SSL_KEYSTORE_LOCATION", "/var/private/ssl/kafka.client.keystore")
                 .withEnv("SCHEMA_REGISTRY_KAFKASTORE_SSL_KEYSTORE_PASSWORD", "confluent")

